src/solver/naive_solver.rs

```diff
@@ -0,0 +1,208 @@
+use crate::models::solution::Solution;
+use crate::models::Order;
+use ethcontract::web3::types::U256;
+
+trait Matchable {
+    /// Returns whether there is a price that satisfies both orders.
+    fn have_price_overlap(&self, other: &Order) -> bool;
+}
+
+impl Matchable for Order {
+    fn have_price_overlap(&self, other: &Order) -> bool {
+        self.get_current_sell_amount() > U256::zero()
+            && other.get_current_sell_amount() > U256::zero()
+            && self.get_current_buy_amount() * other.get_current_buy_amount()
+                <= self.get_current_sell_amount() * other.get_current_sell_amount()
+    }
+}
+
+pub fn solve_pair(sell_orders_token0: &Vec<Order>, sell_orders_token1: &Vec<Order>) -> Solution {
+    assert!(check_orders_sorted_by_limit_price(&sell_orders_token0));
+    assert!(check_orders_sorted_by_limit_price(&sell_orders_token1));
+
+    // Match orders with best limit prices, if possible.
+    if let (Some(best_sell_order_token0), Some(best_sell_order_token1)) =
+        (sell_orders_token0.last(), sell_orders_token1.last())
+    {
+        if best_sell_order_token0.have_price_overlap(&best_sell_order_token1) {
```

Comment:
@josojo, I think I remember you asking how to avoid having this nested `if` statement. There is now a new cool way that was introduced in more recent Rust versions:
```rust
if matches!(
    (sell_orders_token0.last(), sell_orders_token1.last()),
    (Some(best_sell_order_token0), Some(best_sell_order_token1))
    if best_sell_order_token0.have_price_overlap(&best_sell_order_token1)
) {
    // ...
}
```

Although its not very useful here (since you need the values wrapped by the `Option`), just letting you know that this really cool feature was added :smile:.

One other possibility would be to use a `match` statement, this way the compiler guarantees that all paths return the same type:
```rust
match (sell_orders_token0.last(), sell_orders_token1.last()) {
    (Some(best_sell_order_token0), Some(best_sell_order_token1))
        if best_sell_order_token0.have_price_overlap(&best_sell_order_token1) =>
    {
        Solution {
            sell_orders_token0: vec![best_sell_order_token0.clone()],
            sell_orders_token1: vec![best_sell_order_token1.clone()],
        }
    }
    _ => Solution::trivial(),
}
```